src/main/java/homework_4/custom_file_reader/CustomFileReader.java

```diff
@@ -0,0 +1,57 @@
+package homework_4.custom_file_reader;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import static java.lang.System.lineSeparator;
+
+public class CustomFileReader {
+    private static final String FILE_NAME = "src/main/resources/custom_file_reader/Shakespeare.txt";
+    private static final String ERROR_MESSAGE = "Can not read file.";
+    private static final File FILE = new File(FILE_NAME);
+
+    public void run1() {
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILE))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                printMessage(trimDotsAndCommas(line) + lineSeparator());
+            }
+        } catch (IOException e) {
+            printMessage(ERROR_MESSAGE);
+        }
+    }
+
+    public void run2() {
+        try {
```

Comment:
Scanner is not closed here.
```suggestion
try (Scanner scanner = new Scanner(FILE)) {
            while (scanner.hasNextLine()) {
                printMessage(trimDotsAndCommas(scanner.nextLine()) + lineSeparator());
            }
        }
```
or
```suggestion
try {
            Scanner scanner = new Scanner(FILE);
            while (scanner.hasNextLine()) {
                printMessage(trimDotsAndCommas(scanner.nextLine()) + lineSeparator());
            }
            scanner.close();
        }
```

Reply:
thank you, fixed it